llvm/lib/Target/NVPTX/NVPTXLowerArgs.cpp

```diff
@@ -757,3 +746,11 @@ PreservedAnalyses NVPTXCopyByValArgsPass::run(Function &F,
   return copyFunctionByValArgs(F) ? PreservedAnalyses::none()
                                   : PreservedAnalyses::all();
 }
+
+PreservedAnalyses NVPTXLowerArgsPass::run(Function &F,
+                                          FunctionAnalysisManager &AM) {
+  auto &NTM = static_cast<NVPTXTargetMachine &>(TM);
+  bool Changed = isKernelFunction(F) ? runOnKernelFunction(NTM, F)
+                                     : runOnDeviceFunction(NTM, F);
```

Comment:
I might be nice to extract this into a little function do de-duplicate with above.